woocommerce/woocommerce-gateway-stripe
# Fix fatal error in purchasing a subscription with PRB and ECE when the subscription's price or sign-up fee is a string

Fixes #2966 

In `get_product_price` function for PRB/ECE we need `$product->get_price()`, `wc_get_price_including_tax`, `wc_get_price_excluding_tax`, `WC_Subscriptions_Product::get_sign_up_fee` for different scenarios. These functions may return the result in different types as per the corresponding function definitions `int`/`float`/`string`. 

When the types are different the following code causes a fatal error-

```
$product_price = $product->get_price() + WC_Subscriptions_Product::get_sign_up_fee( $product );
```

## Changes proposed in this Pull Request:
- casting the prices to `float` to avoid the fatal error.

## Testing instructions
- The error is not always reproducible. So we should test subscription purchase 
- Create a few products if you don't have them already-
    - A subscription product with no signup fee
    - A variable subscription product with a signup fee
    - A simple subscription product with a signup fee
- Purchase these type of products with PRB button. Confirm that there is no fatal error.
- Enable the ECE feature flag.
- Purchase these type of products with ECE button. Confirm that there is no fatal error.
